File: fleet-tracker/src/main/java/com/hebee/fleet_tracker/service/UserService.java
package com.hebee.fleet_tracker.service;

import java.util.List;

import com.hebee.fleet_tracker.dto.user.UpdatePasswordRequest;
import com.hebee.fleet_tracker.dto.user.UpdateUserRequest;
import com.hebee.fleet_tracker.dto.user.UserRequest;
import com.hebee.fleet_tracker.dto.user.UserResponse;

public interface UserService {

    List<UserResponse> getAllUsers();

    UserResponse getUserById(Long id);

    UserResponse createUser(UserRequest request);

    UserResponse updateUser(Long id, UpdateUserRequest request);

    void updateUserStatus(Long id, Boolean active);

    void updatePassword(Long id, UpdatePasswordRequest request);
}

File: fleet-tracker-ui/src/services/userService.js
import api from "../api/axiosConfig";

const USER_API = "/users";

const userService = {

    getAllUsers: () => api.get(USER_API),

    getUserById: (id) => api.get(`${USER_API}/${id}`),

    createUser: (user) => api.post(USER_API, user),

    updateUser: (id, user) =>
        api.put(`${USER_API}/${id}`, user),

    updateStatus: (id, active) =>
        api.patch(`${USER_API}/${id}/status?active=${active}`),

    updatePassword: (id, password) =>
        api.put(`${USER_API}/${id}/password`, {
            newPassword: password
        })
};

export default userService;